test/units/testsuite-04.sh

```diff
@@ -141,4 +141,87 @@ sleep 3
 # https://github.com/systemd/systemd/issues/15528
 journalctl --follow --file=/var/log/journal/*/* | head -n1 || [[ $? -eq 1 ]]
 
+function add_logs_filtering_override() {
+    UNIT=$1
+    OVERRIDE_NAME=$2
+    LOG_FILTER=${3:-""}
+
+    mkdir -p /etc/systemd/system/"$UNIT".d/
+    echo "[Service]" > /etc/systemd/system/logs-filtering.service.d/"${OVERRIDE_NAME}".conf
+    echo "LogFilterPatterns=$LOG_FILTER" >> /etc/systemd/system/logs-filtering.service.d/"${OVERRIDE_NAME}".conf
+    systemctl daemon-reload
+}
+
+function run_service_and_fetch_logs() {
+    UNIT=$1
+
+    START=$(date '+%Y-%m-%d %T.%6N')
+    systemctl restart "$UNIT"
+    sleep .5
```

Comment:
I'm afraid that this will be brittle. In some CIs we often have large stalls, and I doubt that .5 s will be enough to reliably have something happen and also journald to react to it. If this doesn't cause problems, then OK, but I wouldn't be surprised if it starts popping up in CI failures.

Reply:
I have had no issues so far, even with CI. I could bump it to 1s, although this would just be a workaround. I could also use `systemctl restart --wait` to ensure the unit completes (meaning it echo'ed already), but there is no guarantee `journalctl` would pick up the log entry before the unit's cgroup is removed.